src/obj.rs

```diff
@@ -0,0 +1,4 @@
+trait Object {
+  pub fn new(Vec3D center) -> Self;
+  pub fn intersect(Vec3D dir) -> Option<Color>;
```

Comment:
This signature should be something like `pub fn intersect(Vec3D origin, Vec3D dir) -> Option<f64>`

Note that I also changed the return value to Option<f64>. This represents the distance that the ray travels before it intersects with the object. We will call the intersect function for many objects, and we want to find the closest object.